### PR TITLE
Fieldset: Accordion Mode

### PR DESCRIPTION
**Pull request by ElliotTheHuman**
### :sparkles: Changes

- Added `accordion` prop and accordion-related props to `Fieldset`
- Added additional tests and documentation related to accordion mode on `Fieldset`
- Refactored `Accordion`'s interface structure to better suit extension

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] PR is ideally < ~400LOC

### :camera: Screenshots
Screen Shot 2020-06-04 at 4 27 52 PM

### CODE REVIEW

File: packages/www/src/documentation/components/forms/fieldset.mdx [hunk 1]
```diff
@@ -59,3 +59,52 @@ Using `label` as a props in `Field`, gives a larger clickable area since the are
...
+For more details on all available open state control and indicator styling props for `Accordion`, visit the [Accordion documentation](https://components.looker.com/components/content/accordion).
+
+```jsx
+<Fieldset
+  legend="This is the Legend"
+  accordion
+  defaultOpen
+  onOpen={() => alert('Opening the pod bay doors')}
+  onClose={() => alert('Closing the pod bay doors')}
+  indicatorGap="xxlarge"
```
Comment (lukelooker) [inline packages/www/src/documentation/components/forms/fieldset.mdx:104]: This doesn't do anything, correct? 

```suggestion
```

File: packages/components/src/Form/Fieldset/Fieldset.tsx [hunk 2]
```diff
@@ -27,25 +27,60 @@
...
+    } = omit(props, [...AccordionControlPropKeys])
+    const accordionIndicatorDefaults: AccordionIndicatorProps = {
+      indicatorGap: 'xsmall',
+      indicatorIcons: {
+        close: 'ArrowRight',
+        open: 'ArrowDown',
+      },
+      indicatorPosition: 'left',
+      indicatorSize: 'medium',
+    }
```
Comment (lukelooker) [inline packages/components/src/Form/Fieldset/Fieldset.tsx:79]: Since this isn't dynamic it can be declared outside of the component.

Comment (ElliotTheHuman) [reply packages/components/src/Form/Fieldset/Fieldset.tsx]: Resolved in c55be0a!

File: packages/www/src/documentation/components/forms/fieldset.mdx (hunk 1 above)
Comment (ElliotTheHuman) [reply packages/www/src/documentation/components/forms/fieldset.mdx]: Resolved in c55be0a!

**Pull request merged**